examples/ovault-evm/deploy/MyOvault.ts

```diff
@@ -69,51 +79,80 @@ const deploy: DeployFunction = async (hre) => {
         // 🎯 Get asset address (existing or deployed)
         let assetOFTAddress: string
 
-        if (DEPLOYMENT_CONFIG.vault.assetAddress) {
-            assetOFTAddress = DEPLOYMENT_CONFIG.vault.assetAddress
+        if (DEPLOYMENT_CONFIG.vault.assetOFTAddress) {
+            assetOFTAddress = DEPLOYMENT_CONFIG.vault.assetOFTAddress
             console.log(`Using existing asset address: ${assetOFTAddress}`)
         } else {
             // Use deployed address or get from deployments
             assetOFTAddress =
-                deployedContracts.assetOFT || (await hre.deployments.get(DEPLOYMENT_CONFIG.asset.contract)).address
+                deployedContracts.assetOFT || (await hre.deployments.get(DEPLOYMENT_CONFIG.AssetOFT.contract)).address
             console.log(`Using deployed asset address: ${assetOFTAddress}`)
+            // Fetch underlying ERC20 token address from the OFT using the IOFT artifact
+            const IOFTArtifact = await hre.artifacts.readArtifact('IOFT')
+            const oftContract = await hre.ethers.getContractAt(IOFTArtifact.abi, assetOFTAddress)
+            const assetTokenAddress = await oftContract.token()
+            console.log(`Underlying ERC20 token address: ${assetTokenAddress}`)
```

Comment:
Probably make this something like? 
```suggestion
            console.log(`Underlying ERC20 token address found from OFT deployment: ${assetTokenAddress}`)
```